src/RecommendedModule/RecommendedModuleCollection.php

```diff
@@ -0,0 +1,173 @@
+    {
+        foreach ($this->recommendedModules as $recommendedModule) {
+            if ($moduleName === $recommendedModule->getModuleName()) {
+                return $recommendedModule;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRecommendedModuleNames()
+    {
+        $recommendedModuleNames = [];
+
+        foreach ($this->recommendedModules as $recommendedModule) {
+            $recommendedModuleNames[] = $recommendedModule->getModuleName();
+        }
+
+        return $recommendedModuleNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetExists($offset)
+    {
+        return array_key_exists($offset, $this->recommendedModules);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetGet($offset)
+    {
+        return $this->recommendedModules[$offset];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetSet($offset, $value)
+    {
+        $this->recommendedModules[$offset] = $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function offsetUnset($offset)
+    {
+        unset($this->recommendedModules[$offset]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator()
+    {
+        return new ArrayIterator($this->recommendedModules);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        return count($this->recommendedModules);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEmpty()
+    {
+        return empty($this->recommendedModules);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sortByPosition()
+    {
+        uasort($this->recommendedModules, function (RecommendedModuleInterface $recommendedModuleA, RecommendedModuleInterface $recommendedModuleB) {
+            if ($recommendedModuleA->getPosition() === $recommendedModuleB->getPosition()) {
+                return 0;
+            }
+
+            return ($recommendedModuleA->getPosition() < $recommendedModuleB->getPosition()) ? -1 : 1;
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInstalled()
+    {
+        $recommendedModulesInstalled = new static();
+        $recommendedModulesInstalled->recommendedModules = array_filter($this->recommendedModules, function (RecommendedModuleInterface $recommendedModule) {
+            return $recommendedModule->isInstalled();
+        });
+        $recommendedModulesInstalled->sortByPosition();
+
+        return $recommendedModulesInstalled;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getNotInstalled()
+    {
+        $recommendedModulesNotInstalled = new static();
+        $recommendedModulesNotInstalled->recommendedModules = array_filter($this->recommendedModules, function (RecommendedModuleInterface $recommendedModule) {
+            return !$recommendedModule->isInstalled();
```

Comment:
would it be possible to refactor the methods getInstalled / getNotInstalled with just the return filter that changes

Reply:
In fact I can use ArrayCollection of Doctrine instead of a strongly typed collection and use https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/ArrayCollection.php#L328

But this collection is strongly typed and never reused in others projects, so it's maybe overkill.
As I don't know if this will be reused in next version, depends on new API specs...